lib/ansible/module_utils/common/validation.py

```diff
@@ -60,8 +64,20 @@ def check_mutually_exclusive(terms, module_parameters):
         if count > 1:
             results.append(check)
 
-    if results:
-        full_list = ['|'.join(check) for check in results]
+    value_check = []
+    for check_terms in results:
+        # Create pairs of each value for parameters that are mutually exclusive
+        param_combos = combinations((module_parameters.get(k) for k in check_terms), 2)
+
+        # Test each value pair to ensure they evaluate to False
+        for combo in param_combos:
+            if all(combo):
+                value_check.append(check_terms)
+                # If one value pair evaluates to True, then no need to evaluate further combinations
+                break
```

Comment:
I could use feedback to make sure the logic here is correct.

Reply:
i think the value check is incorrect, it should check 'any 2 values are not default' instead of thruthyness